tool/tsh/common/proxy.go

```diff
@@ -556,7 +556,11 @@ func loadAppCertificateWithAppLogin(cf *CLIConf, tc *libclient.TeleportClient, a
 // Returns tuple (certificate, needLogin, err).
 // The boolean `needLogin` will be true if the error returned should go away with successful `tsh app login <appName>`.
 func loadAppCertificate(tc *libclient.TeleportClient, appName string) (certificate tls.Certificate, needLogin bool, err error) {
-	key, err := tc.LocalAgent().GetKey(tc.SiteName, libclient.WithAppCerts{})
+	profile, err := tc.ProfileStatus()
+	if err != nil {
+		return tls.Certificate{}, false, trace.Wrap(err)
+	}
+	key, err := tc.LocalAgent().GetKey(profile.Cluster, libclient.WithAppCerts{})
```

Comment:
Fixed - `tsh app config --cluster=leaf` was returning the cert path for root cluster instead of leaf cluster. I made the wrong change here initially (and in app login), saving the cert to the root path instead of fixing `tsh app config` to use the leaf path.